modules/solace/solace.go

```diff
@@ -0,0 +1,291 @@
+			ports = append(ports, portStr)
+			portsMap[portStr] = struct{}{}
+		}
+	}
+	if len(s.exposedPorts) > 0 {
+		ports = append(ports, s.exposedPorts...)
+	}
+
+	// --- CLI script generation for queue/topic config ---
+	// Reference: https://docs.solace.com/Admin-Ref/CLI-Reference/VMR_CLI_Commands.html
+	var cliScript string
+	if len(s.queues) > 0 {
+		cliScript += "enable\nconfigure\n"
+
+		// Create VPN if not default
+		if s.Vpn != DefaultVpn {
+			cliScript += fmt.Sprintf("create message-vpn %s\n", s.Vpn)
+			cliScript += "no shutdown\n"
+			cliScript += "exit\n"
+			cliScript += fmt.Sprintf("client-profile default message-vpn %s\n", s.Vpn)
+			cliScript += "message-spool\n"
+			cliScript += "allow-guaranteed-message-send\n"
+			cliScript += "allow-guaranteed-message-receive\n"
+			cliScript += "allow-guaranteed-endpoint-create\n"
+			cliScript += "allow-guaranteed-endpoint-create-durability all\n"
+			cliScript += "exit\n"
+			cliScript += "exit\n"
+			cliScript += fmt.Sprintf("message-spool message-vpn %s\n", s.Vpn)
+			cliScript += "max-spool-usage 60000\n"
+			cliScript += "exit\n"
+		}
+
+		// Configure username and password
+		cliScript += fmt.Sprintf("create client-username %s message-vpn %s\n", s.Username, s.Vpn)
+		cliScript += fmt.Sprintf("password %s\n", s.Password)
+		cliScript += "no shutdown\n"
+		cliScript += "exit\n"
+
+		// Configure VPN Basic authentication
+		cliScript += fmt.Sprintf("message-vpn %s\n", s.Vpn)
+		cliScript += "authentication basic auth-type internal\n"
+		cliScript += "no shutdown\n"
+		cliScript += "end\n"
+
+		// Configure queues and topic subscriptions
+		cliScript += "configure\n"
+		cliScript += fmt.Sprintf("message-spool message-vpn %s\n", s.Vpn)
+		for queue, topics := range s.queues {
+			// Create the queue first
+			cliScript += fmt.Sprintf("create queue %s\n", queue)
+			cliScript += "access-type exclusive\n"
+			cliScript += "permission all consume\n"
+			cliScript += "no shutdown\n"
+			cliScript += "exit\n"
+
+			// Add topic subscriptions to the queue
+			for _, topic := range topics {
+				cliScript += fmt.Sprintf("queue %s\n", queue)
+				cliScript += fmt.Sprintf("subscription topic %s\n", topic)
+				cliScript += "exit\n"
+			}
+		}
+		cliScript += "exit\nexit\n"
+	}
+
+	req := testcontainers.ContainerRequest{
+		Image:        s.Image,
+		ExposedPorts: ports,
+		Env:          env,
+		Cmd:          nil,
+		WaitingFor:   wait.ForHTTP("/").WithPort("8080/tcp").WithStartupTimeout(1 * time.Minute),
+		HostConfigModifier: func(hc *container.HostConfig) {
+			hc.ShmSize = shmSize
+		},
+	}
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		return err
+	}
+
+	s.Container = container // assign before using s.Container
+
+	// Wait for Solace broker to be fully active before configuring
+	if err := s.waitForSolaceActive(ctx); err != nil {
```

Comment:
suggestion: you should use the wait package in the testcontainers core project, to add it in the WaitingFor field of the container request